src/aggregate/expr/expression.c

```diff
@@ -152,11 +155,14 @@ static int evalPredicate(ExprEval *eval, const RSPredicate *pred, RSValue *resul
   RSValue *l_ptr = RSValue_Dereference(&l);
   RSValue *r_ptr = RSValue_Dereference(&r);
 
-  int res = getPredicateBoolean(&l, &r, pred->cond);
-
-  result->numval = res;
-  result->t = RSValue_Number;
-  rc = EXPR_EVAL_OK;
+  int res = getPredicateBoolean(eval, &l, &r, pred->cond);
+  if (!eval->err || eval->err->code == QUERY_OK) {
```

Comment:
Wouldn't this might change the behavior?

Reply:
It probably will, but I don't see any other way to report an error coming out of getPredicateBoolean(). If we don't return EXPR_EVAL_ERR, the error will disappear, and if we don't set RSValue_Undef, we've returned a number that isn't there.